common/service/dynamicconfig/config.go

```diff
@@ -157,346 +166,354 @@ func (c *Collection) GetProperty(key Key, defaultValue interface{}) PropertyFn {
 	return func(domain string) string {
-		filters := append([]FilterOption{DomainFilter(domain)}, c.filterOptions...)
+		filters := c.getFilterMap(DomainFilter(domain))
 		val, err := c.client.GetStringValue(
 			key,
-			getFilterMap(filters...),
+			filters,
 			defaultValue,
 		)
 		if err != nil {
-			c.logError(key, err)
+			c.logError(key, filters, err)
 		}
-		c.logValue(key, val, defaultValue, stringCompareEquals)
+		c.logValue(key, filters, val, defaultValue, stringCompareEquals)
 		return val
 	}
 }
 
 // GetBoolPropertyFilteredByDomain gets property with domain filter and asserts that it's a bool
 func (c *Collection) GetBoolPropertyFilteredByDomain(key Key, defaultValue bool) BoolPropertyFnWithDomainFilter {
 	return func(domain string) bool {
-		filters := append([]FilterOption{DomainFilter(domain)}, c.filterOptions...)
+		filters := c.getFilterMap(DomainFilter(domain))
 		val, err := c.client.GetBoolValue(
 			key,
-			getFilterMap(filters...),
+			filters,
 			defaultValue,
 		)
 		if err != nil {
-			c.logError(key, err)
+			c.logError(key, filters, err)
 		}
-		c.logValue(key, val, defaultValue, boolCompareEquals)
+		c.logValue(key, filters, val, defaultValue, boolCompareEquals)
 		return val
 	}
 }
 
 // GetBoolPropertyFilteredByDomainID gets property with domainID filter and asserts that it's a bool
 func (c *Collection) GetBoolPropertyFilteredByDomainID(key Key, defaultValue bool) BoolPropertyFnWithDomainIDFilter {
 	return func(domainID string) bool {
-		filters := append([]FilterOption{DomainIDFilter(domainID)}, c.filterOptions...)
+		filters := c.getFilterMap(DomainIDFilter(domainID))
 		val, err := c.client.GetBoolValue(
 			key,
-			getFilterMap(filters...),
+			filters,
 			defaultValue,
 		)
 		if err != nil {
-			c.logError(key, err)
+			c.logError(key, filters, err)
 		}
-		c.logValue(key, val, defaultValue, boolCompareEquals)
+		c.logValue(key, filters, val, defaultValue, boolCompareEquals)
 		return val
 	}
 }
 
 // GetBoolPropertyFilteredByTaskListInfo gets property with taskListInfo as filters and asserts that it's an bool
 func (c *Collection) GetBoolPropertyFilteredByTaskListInfo(key Key, defaultValue bool) BoolPropertyFnWithTaskListInfoFilters {
 	return func(domain string, taskList string, taskType int) bool {
-		filters := append(
-			[]FilterOption{
-				DomainFilter(domain),
-				TaskListFilter(taskList),
-				TaskTypeFilter(taskType),
-			},
-			c.filterOptions...,
+		filters := c.getFilterMap(
+			DomainFilter(domain),
+			TaskListFilter(taskList),
+			TaskTypeFilter(taskType),
 		)
 		val, err := c.client.GetBoolValue(
 			key,
-			getFilterMap(filters...),
+			filters,
 			defaultValue,
 		)
 		if err != nil {
-			c.logError(key, err)
+			c.logError(key, filters, err)
 		}
-		c.logValue(key, val, defaultValue, boolCompareEquals)
+		c.logValue(key, filters, val, defaultValue, boolCompareEquals)
 		return val
 	}
 }
+
+func (c *Collection) getFilterMap(opts ...FilterOption) map[Filter]interface{} {
```

Comment:
Nit: toFilterMap(...) GetXxx is generally used when it is got from c *Collection, while this is converting the input paramters into another object